pkg/registry/core/service/strategy_test.go

```diff
@@ -1053,6 +1103,18 @@ func TestDropTypeDependentFields(t *testing.T) {
 			if result.Spec.HealthCheckNodePort != tc.expect.Spec.HealthCheckNodePort {
 				t.Errorf("failed %q: expected healthCheckNodePort %d, got %d", tc.name, tc.expect.Spec.HealthCheckNodePort, result.Spec.HealthCheckNodePort)
 			}
+			if tc.expect.Spec.AllocateLoadBalancerNodePorts == nil {
```

Comment:
can we just do a `reflect.DeepEqual` with `tc.expect.Spec.AllocateLoadBalancerNodePorts` and `result.Spec.AllocateLoadBalancerNodePorts`?

Reply:
Yes, better! But error printouts have pointers (provoked fail);
```
go test
--- FAIL: TestDropTypeDependentFields (0.00s)
    --- FAIL: TestDropTypeDependentFields/set_allocatedLoadBalancerNodePorts_false_->_nil (0.00s)
        strategy_test.go:1107: failed "set allocatedLoadBalancerNodePorts false -> nil": expected AllocateLoadBalancerNodePorts 0xc000530456, got <nil>
FAIL
exit status 1
FAIL    k8s.io/kubernetes/pkg/registry/core/service     0.017s
```